toolchain/semantics/semantics_context.cpp

```diff
@@ -143,6 +224,20 @@ auto SemanticsContext::AddCurrentCodeBlockToFunction() -> void {
       .body_block_ids.push_back(node_block_stack().PeekForAdd());
 }
 
+auto SemanticsContext::CurrentPositionIsReachable() -> bool {
+  if (!node_block_stack().CurrentBlockIsReachable()) {
+    return false;
+  }
+
+  // Our current position is at the end of the current block. That position is
+  // reachable unless the previous instruction is a terminator instruction.
+  auto& block_contents = semantics_ir().GetNodeBlock(node_block_stack().Peek());
```

Comment:
Peek is not guaranteed to provide a valid return. This probably works today because it won't be called from a lazily created block. I'd suggest one of:

1) Document in the header that this can only be called when a valid block is on the top. Maybe explicitly CHECK, but GetNodeBlock will fail if it gets an invalid block id.
2) Handle invalid. Probably `auto block_id = node_block_stack().Peek(); if (!block_id.is_valid()) { return true; }`

My slight leaning is option 1 because it's less logic, and the guard rail can be switched if behavior changes.

Reply:
Nice catch. This is working today because code blocks are never created as invalid -- we eagerly create the entry block so we can add it to the `SemanticsFunction`, and we eagerly create all other code blocks so we can create a branch to them. But I think it'd probably be reasonable to call this function after a new code block is pushed and before any branches to it are added -- and at least it'd be surprising if it fails in that case -- so I've gone with option 2.